pkg/skaffold/parser/config.go

```diff
@@ -165,7 +165,7 @@ func processEachConfig(config *latest_v1.SkaffoldConfig, cfgOpts configOpts, opt
 		return nil, sErrors.ConfigSetDefaultValuesErr(config.Metadata.Name, cfgOpts.file, err)
 	}
 	// convert relative file paths to absolute for all configs that are not invoked explicitly. This avoids maintaining multiple root directory information since the dependency skaffold configs would have their own root directory.
-	if cfgOpts.isDependency {
+	if cfgOpts.isDependency || opts.MakePathsAbsolute {
```

Comment:
For root level `skaffold.yaml`, `cfgOpts.file == "-"` is possible when reading from STDIN. That wasn't a case before since `cfgOpts.isDependency` was false for root level `skaffold.yaml` files. 
So I think you need an additional check here, just test on some project with `cat skaffold.yaml | skaffold -f - --yaml-only`

Reply:
sorry not sure i completely understand, what's the failure case here? the change i'm making will force absolute path resolution for all runs of `skaffold diagnose`, but won't change the behavior for other command invocations. so this won't have any effect on `dev`, `run`, etc. it seems to work fine when i test it out on the `multi-config-microservices` project

Reply:
actually I'm sorry, I should have run what I typed out instead of just relying on what I thought would happen 😅

I thought that `parser.GetAllConfigs` couldn't handle `-f -` for setting absolute paths to current directory. It does however handle that correctly. My bad.

Reply:
even though I wrote that code 🤷🏾‍♂️🤦🏽‍♂️